File: SwiftBridge/Sources/SwiftBridge/BasicPlatform.swift
//
//  File.swift
//  
//
//  Created by Luke Kaalim on 26/12/2022.
//

import Foundation
import QuickJS

public struct BasicPlatform: Platform {
    public let console = ConsoleService();
    public let render = RenderService();
    public let timeout = TimeoutService();
    
    public init() {
        
    }
    
    public func CreatePlatformObject(context: JSContext) -> JSValue {
        let object = context.createObject();
        object.setProperty("render", render.CreateServiceObject(context: context));
        object.setProperty("console", console.CreateServiceObject(context: context));
        object.setProperty("timeout", timeout.CreateServiceObject(context: context));
        object.dup();
        
        return object;
    }
}

File: KotlinBridge/KotlinBridge/src/main/java/com/lukekaalim/kotlinbridge/BasicPlatform.kt
package com.lukekaalim.kotlinbridge

import android.content.Context
import android.view.ViewGroup
import com.lukekaalim.kotlinbridge.Services.Console
import com.lukekaalim.kotlinbridge.Services.Renderer
import com.lukekaalim.kotlinbridge.Services.Timeout
import com.whl.quickjs.wrapper.JSObject
import com.whl.quickjs.wrapper.QuickJSContext

class BasicPlatform(val context: Context, val rootView: ViewGroup): Platform {
    val render = Renderer(context, rootView);
    val console = Console();
    val timeout = Timeout();

    override fun createPlatformObject(context: QuickJSContext): JSObject {
        val obj = context.createNewJSObject();
        obj.setProperty("render", render.createServiceObject(context));
        obj.setProperty("console", console.createServiceObject(context));
        obj.setProperty("timeout", timeout.createServiceObject(context));
        return obj;
    }
}